attestor/root.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pkg defines fns for running Scorecard checks on a Repo.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/golang/glog"
+	"github.com/grafeas/kritis/pkg/attestlib"
+	"github.com/grafeas/kritis/pkg/kritis/metadata/containeranalysis"
+	"github.com/grafeas/kritis/pkg/kritis/signer"
+	"github.com/grafeas/kritis/pkg/kritis/util"
+	"github.com/ossf/scorecard-attestor/policy"
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/checks"
+	sclog "github.com/ossf/scorecard/v4/log"
+	"github.com/ossf/scorecard/v4/pkg"
+)
+
+type SignerMode string
+
+const (
+	CheckAndSign  SignerMode = "check-and-sign"
+	CheckOnly     SignerMode = "check-only"
+	BypassAndSign SignerMode = "bypass-and-sign"
+)
+
+var (
+	// input flags
+	repoURL            string
+	commitSHA          string
+	mode               string
+	image              string
+	policyPath         string
+	attestationProject string
+	overwrite          bool
+	noteName           string
+	// input flags: pgp key flags
+	pgpPriKeyPath string
+	pgpPassphrase string
+	// pkix key flags
+	pkixPriKeyPath string
+	pkixAlg        string
+
+	// input flags: kms flags
+	kmsKeyName   string
+	kmsDigestAlg string
+
+	// helper global variables
+	modeFlags   *flag.FlagSet
+	modeExample string
+)
+
+func init() {
+	// need to add all flags to avoid "flag not provided error"
+	addBasicFlags(flag.CommandLine)
+	addCheckFlags(flag.CommandLine)
+	addSignFlags(flag.CommandLine)
+}
+
+func addBasicFlags(fs *flag.FlagSet) {
+	fs.StringVar(&mode, "mode", "check-and-sign", "(required) mode of operation, check-and-sign|check-only|bypass-and-sign")
+	fs.StringVar(&image, "image", "", "(required) image url, e.g., gcr.io/foo/bar@sha256:abcd")
+	fs.StringVar(&repoURL, "repoURL", "", "(required) repo URL from which source was built")
+	fs.StringVar(&commitSHA, "commit", "", "git SHA at which image was built")
+}
+
+func addCheckFlags(fs *flag.FlagSet) {
+	fs.StringVar(&policyPath, "policy", "", "(required for check) scorecard attestation policy file path, e.g., /tmp/policy-binauthz.yml")
+}
+
+func addSignFlags(fs *flag.FlagSet) {
+	fs.StringVar(&noteName, "note_name", "", "(required for sign) note name that created attestations are attached to, in the form of projects/[PROVIDER_ID]/notes/[NOTE_ID]")
```

Comment:
`check-and-sign` and `note_name` use different naming convention `_` vs `-` vs camel case. use one consistently